components/modal/Modal.tsx

```diff
@@ -0,0 +1,81 @@
+import React, { useCallback, useEffect, useRef } from 'react';
+import {
+  ButtonIconRounded,
+  EIConTypes,
+} from '@ost-cas-fee-adv-23-24/elbmum-design';
+import useModal from '@/hooks/useModal';
+
+const escapeKeyCode = 27;
+const Modal = () => {
+  const { content, isOpen, closeModal, title, fullWidth } = useModal();
+  const positionCommon = 'top-0 right-0 bottom-0 left-0';
+  const transitionOptions = 'opacity duration-300 ease delay-300';
+
+  const escapeKeyUpHandler = useCallback(
+    (evt: KeyboardEvent) => {
+      evt.preventDefault();
+      if (evt.keyCode === escapeKeyCode) {
+        closeModal();
+      }
+    },
+    [closeModal],
+  );
+
+  useEffect(() => {
+    if (isOpen) {
+      document.body.addEventListener('keyup', escapeKeyUpHandler);
+    } else {
+      document.body.removeEventListener('keyup', escapeKeyUpHandler);
```

Comment:
is it also working if you use the removeEventListener to the return statement of the useEffect?

Reply:
Do you mean to clean up after demounting the component? 
As the Modal is wrapped in the provider, it would not be demounted.  But we could add an return on the useEffect to clean everything up. 
What do you think?